invalid memory address or nil pointer dereference

<!--
## In what area(s)?
Remove the '> ' to select
> /area core
> /area discovery
> /area interceptor
> /area monitoring
> /area sink
> /area source
-->

## What version of Loggie?
v1.2.0

## Expected Behavior
修复空指针问题


## Actual Behavior
空指针异常
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x813941]

goroutine 293 [running]:
github.com/loggie-io/loggie/pkg/pipeline.(*Pipeline).finalizeBatch(0xc00051e140, {0x258bfe0, 0xc000fada00})
	/pkg/pipeline/pipeline.go:345 +0x161
github.com/loggie-io/loggie/pkg/pipeline.(*Pipeline).consumerOutChanAndDrop(0xc00051e140, 0xc000332b40)
	/pkg/pipeline/pipeline.go:182 +0x145
created by github.com/loggie-io/loggie/pkg/pipeline.(*Pipeline).cleanOutChan
	/pkg/pipeline/pipeline.go:169 +0xc5
```


## Steps to Reproduce the Problem

<!-- How can a maintainer reproduce this issue (be detailed) -->

--- a/pkg/pipeline/pipeline.go
+++ b/pkg/pipeline/pipeline.go
@@ -173,16 +173,23 @@ func (p *Pipeline) cleanOutChan() {
 func (p *Pipeline) consumerOutChanAndDrop(out chan api.Batch) {
 	after := time.NewTimer(p.config.CleanDataTimeout)
 	defer after.Stop()
+	dropAndRelease := func(batch api.Batch) {
+		if batch != nil {
+			events := batch.Events()
+			if events != nil {
+				p.info.EventPool.PutAll(events)
+			}
+			batch.Release()
+		}
+	}
 	for {
 		select {
 		case <-after.C:
 			return
 		case b := <-out:
-			// drop
-			p.finalizeBatch(b)
+			dropAndRelease(b)
 		case b := <-p.info.SurviveChan:
-			// drop
-			p.finalizeBatch(b)
+			dropAndRelease(b)
 		}
 	}
 }

